ios/src/main/java/org/ergoplatform/ios/api/IosKeychainAccess.kt

```diff
@@ -0,0 +1,98 @@
+package org.ergoplatform.ios.api
+
+import org.ergoplatform.utils.LogUtils
+import org.robovm.apple.corefoundation.CFData
+import org.robovm.apple.corefoundation.OSStatusException
+import org.robovm.apple.foundation.NSString
+import org.robovm.apple.foundation.NSStringEncoding
+import org.robovm.apple.localauthentication.LAContext
+import org.robovm.apple.security.*
+
+object IosKeychainAccess {
+    // https://developer.apple.com/documentation/localauthentication/accessing_keychain_items_with_face_id_or_touch_id
+    // https://medium.com/@alx.gridnev/biometry-protected-entries-in-ios-keychain-6125e130e0d5
+
+    private const val ERR_KEY_NOT_EXISTING = -25300
+    private const val ERR_USER_CANCELLED = -128
+    // returned when key is rendered inaccessible because user deactivated device authentication
+    private const val ERR_NOT_AVAILABLE = -25291
+    private const val ERR_AUTH_FAILED = -25293
+
+    fun savePassword(service: String, name: String, value: String) {
+        val attributes = SecAttributes().apply {
+            set(SecQuery.Keys.Class(), SecClass.Values.GenericPassword())
+            setAccount(name)
+            setService(service)
+            set(SecValue.Keys.Data(), NSString(value).toData(NSStringEncoding.UTF8))
+            setAccessControl(
+                SecAccessControl.create(
+                    SecAttrAccessible.WhenPasscodeSetThisDeviceOnly,
+                    SecAccessControlCreateFlags.UserPresence
+                )
+            )
+        }
+        try {
+            SecItem.add(attributes)
+        } catch (t: Throwable) {
+            LogUtils.logDebug("IosKeychainAccess", "Could not add key", t)
+            // throw this so user knows this wasn't saved
+            throw t
+        }
+    }
+
+    fun deletePassword(service: String, name: String) {
+        val query = SecQuery().apply {
+            set(SecQuery.Keys.Class(), SecClass.Values.GenericPassword())
+            attributes = SecAttributes().apply {
+                setAccount(name)
+                setService(service)
+            }
+        }
+
```

Comment:
why setUseAuthenticationContext(context) is not called in this method?

Reply:
It's not needed to delete the secret, which is important since the app must be able to delete the secret when it is inaccessible (for example when user deactivated authentication after saving the secret)